prog/vnet/subnet_nexus.rb

```diff
@@ -175,11 +175,12 @@ def self.random_private_ipv6(location, project)
 
   def self.random_private_ipv4(location, project)
     private_range = PrivateSubnet.random_subnet
+    banned_subnets = PrivateSubnet::BANNED_IPV4_SUBNETS.map { NetAddr::IPv4Net.parse(_1) }
     addr = NetAddr::IPv4Net.parse(private_range)
 
     selected_addr = addr.nth_subnet(26, SecureRandom.random_number(2**(26 - addr.netmask.prefix_len) - 1).to_i + 1)
 
-    if project.private_subnets.find { _1.net4.to_s == selected_addr.to_s && _1.location == location }
+    if project.private_subnets.find { _1.net4.to_s == selected_addr.to_s && _1.location == location } || banned_subnets.any? { _1.rel(selected_addr) }
```

Comment:
push down into database again? "not exist" with a subquery. LLMs can usually get this right, I always use it rather than cracking open the manual again.

Reply:
what about now?